core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForNoInvertedIndexForShort.java

```diff
@@ -79,12 +82,8 @@ private void rleEncodeOnData(List<byte[]> actualDataList) {
     }
   }
 
-  private byte[][] convertToDataPage(List<byte[]> list) {
-    byte[][] shortArray = new byte[list.size()][];
-    for (int i = 0; i < shortArray.length; i++) {
-      shortArray[i] = list.get(i);
-    }
-    return shortArray;
+  private ByteBuffer[] convertToDataPage(List<ByteBuffer> list) {
```

Comment:
should avoid redundant conversion,  should directly use ByteBuffer[] everywhere, don't convert to list